### PR TITLE
Add cluster label for remote_cluster_sync_timeouts_total metric

### PR DESCRIPTION
**Pull request by kebe7jun**
Fix https://github.com/istio/istio/issues/44489

### CODE REVIEW

File: releasenotes/notes/add-cluster-id-for-cluster-sync-metrics.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+# area describes the area that this change affects.
+# Valid values are:
+# - traffic-management
+# - security
+# - telemetry
+# - installation
+# - istioctl
+# - documentation
+area: traffic-management
```
Comment (zirain) [inline releasenotes/notes/add-cluster-id-for-cluster-sync-metrics.yaml:23]: should be telemetry

File: pkg/kube/multicluster/cluster.go [hunk 2]
```diff
@@ -49,7 +49,7 @@ func (r *Cluster) Run() {
 		time.AfterFunc(features.RemoteClusterTimeout, func() {
 			if !r.initialSync.Load() {
 				log.Errorf("remote cluster %s failed to sync after %v", r.ID, features.RemoteClusterTimeout)
-				timeouts.Increment()
+				timeouts.With(clusterID.Value(string(r.ID))).Increment()
```
Comment (zirain) [inline pkg/kube/multicluster/cluster.go:52]: can we remove/clean the stats when a cluster is removed?

Comment (kebe7jun) [reply pkg/kube/multicluster/cluster.go:52]: `remote_cluster_sync_timeouts_total` is a Metric of type `Sum`, which should count all occurrences of errors, maybe we shouldn't be clear about it? 
When doing alerts based on this metric, you should also use `rate` calculation, it should not affect, what do you think?

Comment (zirain) [reply pkg/kube/multicluster/cluster.go:52]: The worst-case scenario, too much cluster removed and never add back, is there some memory leak?

Comment (zirain) [reply pkg/kube/multicluster/cluster.go:52]: After deep thinking, there're not too much cluster in a real world, and it's safe to rollout istiod if there's a real memory issue.

**Review by zirain — APPROVED** [review]
LGTM

**Comment by howardjohn** [comment]
In other places we use `cluster` instead of `cluster_id` (controller_sync_errors_total).

**Review by jaellio — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

**Pull request merged**